source/write/bulk-write.txt

```diff
@@ -0,0 +1,343 @@
+instance of one of the following operation classes that inherit from the
+generic ``WriteModel`` class:
+
+- ``InsertOneModel``
+- ``UpdateOneModel``
+- ``UpdateManyModel``
+- ``ReplaceOneModel``
+- ``DeleteOneModel``
+- ``DeleteManyModel``
+
+Then, pass a list of these instances to the ``bulkWrite()`` method.
+
+The following sections show how to create and use instances of the
+preceding classes. The :ref:`scala-bulkwrite-method` section
+demonstrates how to pass a list of models to the ``bulkWrite()`` method
+to perform the bulk operation.
+
+Insert Operations
+~~~~~~~~~~~~~~~~~
+
+To perform an insert operation, create an ``InsertOneModel`` instance and specify
+the document you want to insert.
+
+The following example creates an instance of ``InsertOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :start-after: start-bulk-insert-one
+   :end-before: end-bulk-insert-one
+   :language: scala
+   :copyable:
+   :dedent:
+
+To insert multiple documents, create an instance of ``InsertOneModel``
+for each document.
+
+.. important::
+
+   When performing a bulk operation, the ``InsertOneModel`` cannot
+   insert a document with an ``_id`` that already exists in the
+   collection. In this situation, the driver throws a
+   ``MongoBulkWriteException``.
+
+Update Operations
+~~~~~~~~~~~~~~~~~
+
+To update a document, create an instance of ``UpdateOneModel`` and pass
+the following arguments:
+
+- A **query filter** that specifies the criteria used to match documents in your collection
+- The update operation you want to perform. For more information about update
+  operations, see the :manual:`Field Update Operators
+  </reference/operator/update-field/>` guide in the {+mdb-server+} manual.
+
+An ``UpdateOneModel`` instance specifies an update for *the first*
+document that matches your query filter.
+
+The following example creates an instance of ``UpdateOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :start-after: start-bulk-update-one
+   :end-before: end-bulk-update-one
+   :language: scala
+   :copyable:
+   :dedent:
+
+To update multiple documents, create an instance of ``UpdateManyModel`` and pass
+the same arguments as for ``UpdateOneModel``. The ``UpdateManyModel``
+class specifies updates for *all* documents that match your query
+filter.
+
+The following example creates an instance of ``UpdateManyModel``:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :start-after: start-bulk-update-many
+   :end-before: end-bulk-update-many
+   :language: scala
+   :copyable:
+   :dedent:
+
+Replace Operations
+~~~~~~~~~~~~~~~~~~
+
+A replace operation removes all fields and values of a specified document and
+replaces them with new fields and values that you specify. To perform a
+replace operation, create an instance of ``ReplaceOneModel`` and pass a
+query filter and the fields and values you want to replace the matching
+document with.
```

Comment:
S: not necessary, but you can list the arguments here like you did with the update section:
```suggestion
replace operation, create an instance of ``ReplaceOneModel`` and pass the following arguments:

- Query filter ...
- Replacement document ...
```